modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -5257,6 +5258,9 @@ public void setParameterBooleanFlagWithCodegenProperty(CodegenParameter paramete
      * @param var list of CodegenProperty
      */
     public void updateCodegenPropertyEnum(CodegenProperty var) {
+        if (!var.isPrimitiveType) {
+            return;
```

Comment:
What is the rationale for returning early here?

This prevents the allowableValues/EnumVars from being constructed at all, which I think you'd want to do if it's an Enum in any way.

Reply:
Ah sorry, I should have removed the summoning from description. I gave a bit more context to Jim about that base PR.
first is the actual intended change, the second is a lame attempt at fixing all the changes that result from it. So there is no rationale behind this change, I was just trying stuff out.

So yes, you are absolutely right here. My bad!